### ISSUE TITLE
Search

### ISSUE THREAD

**Issue by PanderMusubi** [intergalacticfm/hotmixes.resty#16]
Add search functionality

**Comment by PanderMusubi** [comment]
Search should also be available as a link, e.g. https://hotmixes.net/search?artistname should lead to a page with all search results. Some artists are only under Panama Racing Club or IFM Fests and can't be found via thei name entry.

**Comment by yrew** [comment]
> Search should also be available as a link, e.g. https://hotmixes.net/search?artistname should lead to a page with all search results. Some artists are only under Panama Racing Club or IFM Fests and can't be found via thei name entry.

Yes, I already implemented it like that. I only need to show the results on the screen. And we need to have a search field somewhere.

**Comment by dromer** [comment]
I see you are passing the search arguments to `popen()`, this has a very likely gaping security hole in it.

Just so you are aware :)

**Comment by yrew** [comment]
> I see you are passing the search arguments to `popen()`, this has a very likely gaping security hole in it.
> 
> Just so you are aware :)

Thanks for pointing that out. Any suggestions on how to this in a more secure way?

**Comment by PanderMusubi** [comment]
Have a look at lfs LuaFileSystem, see e.g. https://stackoverflow.com/questions/5303174/how-to-get-list-of-directories-in-lua/5303802#5303802

**Comment by yrew** [comment]
I did but it's not of any help with searching the filesystem as far as I understand

**Comment by dromer** [comment]
Maybe we need a way to index the filesystem.

Cron job that runs every 10 minutes or so? Doesn't have to be that fine-grained.
Thing is what format and how to query?

**Comment by dromer** [comment]
> Have a look at lfs LuaFileSystem, see e.g. https://stackoverflow.com/questions/5303174/how-to-get-list-of-directories-in-lua/5303802#5303802

Also used LFS to fix this leak: https://github.com/intergalacticfm/hotmixes.resty/commit/f90ef545a3aba97bec5288465f9c9d0137388d19

**Comment by yrew** [comment]
> Maybe we need a way to index the filesystem.
> 
> Cron job that runs every 10 minutes or so? Doesn't have to be that fine-grained.
> Thing is what format and how to query?

I thought about this as well but I am really hoping sanitizing the input before calling popen will make it's use safe enough.
